cpp/oneapi/dal/backend/primitives/ndarray.hpp

```diff
@@ -210,8 +211,52 @@ class ndview : public ndarray_base<axis_count, order> {
         return this->get_count() > 0;
     }
 
+    bool is_data_mutable() const {
```

Comment:
It will confuse class user. You introduce similar method and the difference isn't clear until you look at the code.
I suggest remove it and use has_mutable_data instead.